src/pages/SingleProductPage.js

```diff
@@ -97,13 +97,21 @@ const SingleProductPage = () => {
             
           </section>
           <div>
-            {isAdminLoggedIn && <UpdateProduct product={product} onUpdate={() => fetchSingleProduct(url, id)}/>}
-         
-            {isAdminLoggedIn && <DeleteProduct productId={id}/>}
-            </div>
-<Reviews productId ={id}/>
+            {isAdminLoggedIn && (
+              <UpdateProduct
+                product={product}
+                onUpdate={() => fetchSingleProduct(url, id)}
+              />
+            )}
+
+            {isAdminLoggedIn && <DeleteProduct productId={id} />}
+          </div>
+          <Reviews
+            productId={id}
+            reviews={product.reviews || []}
```

Comment:
Always try to give components just the information they need. So `<Reviews>` doesn't care about the full product object, it only cares about the `product.reviews` bit. 

And to avoid any errors,  product doesn't have any reviews, just fall back to passing in an empty array.